cmd/grype/cli/commands/db_get.go

```diff
@@ -0,0 +1,132 @@
+package commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"strings"
+
+	"github.com/anchore/clio"
+	"github.com/anchore/grype/grype"
+	"github.com/anchore/grype/grype/db"
+	"github.com/anchore/grype/grype/store"
+	"github.com/anchore/grype/grype/vulnerability"
+	"github.com/anchore/grype/internal/bus"
+	"github.com/anchore/grype/internal/log"
+	"github.com/hashicorp/go-multierror"
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+)
+
+type dbQueryOptions struct {
+	Output    string `yaml:"output" json:"output" mapstructure:"output"`
+	DBOptions `yaml:",inline" mapstructure:",squash"`
+}
+
+var _ clio.FlagAdder = (*dbQueryOptions)(nil)
+
+func (c *dbQueryOptions) AddFlags(flags clio.FlagSet) {
+	flags.StringVarP(&c.Output, "output", "o", "format to display results (available=[table, json])")
+}
+
+func DBSearch(app clio.Application) *cobra.Command {
+	opts := &dbQueryOptions{
+		Output:    "table",
+		DBOptions: *dbOptionsDefault(app.ID()),
+	}
+
+	return app.SetupCommand(&cobra.Command{
+		Use:   "search [vulnerability_id]",
+		Short: "get information on a vulnerability from the db",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(_ *cobra.Command, args []string) (err error) {
+			id := args[0]
+			return runDbSearch(opts, id)
+		},
+	}, opts)
+}
+
+func runDbSearch(opts *dbQueryOptions, cveID string) (errs error) {
+	var str *store.Store
+	var status *db.Status
+	var dbCloser *db.Closer
+
+	err := parallel(
+		func() (err error) {
+			log.Debug("loading DB")
+			str, status, dbCloser, err = grype.LoadVulnerabilityDB(opts.DB.ToCuratorConfig(), opts.DB.AutoUpdate)
+			return validateDBLoad(err, status)
+		},
+	)
+
+	if err != nil {
+		return err
+	}
+
+	if dbCloser != nil {
+		defer dbCloser.Close()
+	}
+
+	vulnerabilities, err := str.Get(cveID, "")
+	if err != nil {
+		return err
+	}
+
+	sb := &strings.Builder{}
+	if len(vulnerabilities) == 0 {
+		sb.WriteString("CVE doesn't exist in the DB\n")
+	} else {
+		err := present(opts.Output, vulnerabilities, sb)
+		if err != nil {
+			errs = multierror.Append(errs, err)
+		}
+	}
+
+	bus.Report(sb.String())
+
+	return errs
+}
```

Comment:
Sorry, I should have taken a closer look before. 🤦  This function could be simplified a bit: it doesn't need a single-entry parallel call, everything needs to happen in serial anyway, which simplifies the error handling to not need any multierror append, and the terminology should be updated to match search/vunlerability id. Something like this:

```
func runDbSearch(opts *dbQueryOptions, vulnerabilityID string) error {
	log.Debug("loading DB")
	str, status, dbCloser, err := grype.LoadVulnerabilityDB(opts.DB.ToCuratorConfig(), opts.DB.AutoUpdate)
	err = validateDBLoad(err, status)
	if err != nil {
		return err
	}
	if dbCloser != nil {
		defer dbCloser.Close()
	}

	vulnerabilities, err := str.Get(vulnerabilityID, "")
	if err != nil {
		return err
	}

	sb := &strings.Builder{}
	if len(vulnerabilities) == 0 {
		return fmt.Errorf("Vulnerability doesn't exist in the DB: %s", vulnerabilityID)
	}
	
	err = present(opts.Output, vulnerabilities, sb)
	bus.Report(sb.String())

	return err
}
```

Reply:
fixed

Reply:
fixed